### DISCUSSION TITLE
`MaybeUndefined<T>` vs `Option<Option<T>>`

### DISCUSSION THREAD

**Discussion by onichandame** [async-graphql/async-graphql#906] [Q&A]
I am wondering the consideration behind the design of `MaybeUndefined`, as it seems that `Option<Option>` perfectly achieves the same functionality. Is there anything that makes it necessary to create the custom enum?

**Comment by SadiinsoSnowfall** [comment]
This was done this way because serde can't properly handle `Option<Option<_>>` deserialization.
See https://github.com/serde-rs/serde/issues/1042 and https://github.com/async-graphql/async-graphql/issues/115
